skpro/regression/linear/_glm.py

```diff
@@ -0,0 +1,387 @@
+            intercept by default). Equivalent to statsmodel's (exog).
+
+        y : pandas DataFrame
+            1d array of the endogenous response variable. This array can be 1d or
+            2d. Binomial family models accept a 2d array  with two columns.
+            If supplied each observation is expected to  be [success, failure].
+            Equivalent to statsmodel's (endog).
+
+        Returns
+        -------
+        self : reference to self
+        """
+        if self.add_constant:
+            X = add_constant(X)
+
+        y_col = y.columns
+
+        glm_estimator = GLM(
+            endog=y,
+            exog=X,
+            family=self.family,
+            offset=self.offset,
+            exposure=self.exposure,
+            freq_weights=self.freq_weights,
+            var_weights=self.var_weights,
+            missing=self.missing,
+        )
+
+        self._estimator = (
+            glm_estimator  # does this need to be cloned using some clone method?
+        )
+
+        fitted_glm_model = self._estimator.fit(
+            self.start_params,
+            self.maxiter,
+            self.method,
+            self.tol,
+            self.scale,
+            self.cov_type,
+            self.cov_kwds,
+            self.use_t,
+            self.full_output,
+            self.disp,
+            self.max_start_irls,
+        )
+
+        # forward some parameters to self
+        FITTED_PARAMS_TO_FORWARD = {"glm_estimator_": fitted_glm_model, "y_col": y_col}
+
+        for k, v in FITTED_PARAMS_TO_FORWARD.items():
+            setattr(self, k, v)
+
+        return self
+
+    def _predict(self, X):
+        """Predict labels for data from features.
+
+        State required:
+            Requires state to be "fitted"
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : pandas DataFrame, same length as `X`, with same columns as y in fit
+        """
+        y_column = self.y_col
+        y_pred_series = self.glm_estimator_.predict(X)
+        y_pred = pd.DataFrame(y_pred_series, columns=[y_column])
+
+        return y_pred
+
+    def _predict_proba(self, X):
+        """Predict distribution over labels for data from features.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y_pred : skpro BaseDistribution, same length as `X`
+            labels predicted for `X`
+        """
+        from skpro.distributions.normal import Normal
+
+        y_pred_series = self.glm_estimator_.predict(X)
```

Comment:
That's a sensible variance estimate, but you need to get it from the GLM model - it produces its own varianve predictions!

The method is `get_prediction`, you can get mean and variance from it:
https://www.statsmodels.org/stable/generated/statsmodels.genmod.generalized_linear_model.GLMResults.get_prediction.html#statsmodels.genmod.generalized_linear_model.GLMResults.get_prediction

Reply:
Thanks for that! I wasn't sure how to derive the mu and sigma and I realized I probably should have mentioned that in the main PR req

Reply:
no worries!

That's actually one of the key points that `skpro` means to help the user and deveoper with - where or how exactly the distributions are represented differs wildly by package, and even within packages ilke `sklearn` and `statsmodels`. There is no clear standard, or unified API for distribution predictors out there